test/integration/master_routes_test.go

```diff
@@ -171,6 +175,124 @@ func TestWellKnownOAuthOff(t *testing.T) {
 	}
 }
 
+func TestApiGroups(t *testing.T) {
+	testutil.RequireEtcd(t)
```

Comment:
Split this test function to three function `TestBuildApiGroup ` `TestDeployApiGroup ` `TestTemplateApiGroup `maybe better, and add another common test config function. :)